src/orca/services/nextflowtower/client.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any
+
+import requests
+from pydantic.dataclasses import dataclass
+
+
+@dataclass(kw_only=False)
+class NextflowTowerClient:
+    """Simple Python client for making requests to Nextflow Tower.
+
+    Attributes:
+        api_endpoint: API endpoint for a Nextflow Tower platform.
+        auth_token: An authentication token for the platform specified
+            by the ``api_endpoints`` value.
+    """
+
+    auth_token: str
+    api_endpoint: str
+
+    def update_kwarg(
+        self, kwargs: dict[str, Any], key1: str, key2: str, default: Any
+    ) -> None:
+        """Ensure a default value for a nested key in kwargs.
+
+        Args:
+            kwargs: Keyword arguments
+            key1: Top-level key.
+            key2: Nested key.
+            default: Default value.
+        """
+        kwargs.setdefault(key1, dict())
+        if not isinstance(kwargs[key1], dict):
+            message = f"The '{key1}' keyword argument must be a dictionary."
+            raise ValueError(message)
+        kwargs[key1].setdefault(key2, default)
+        if not isinstance(kwargs[key1][key2], type(default)):
+            message = (
+                f"The value for kwargs['{key1}']['{key2}'] ({kwargs[key1][key2]}) "
+                f"is not the expected type ({type(default)})."
+            )
+            raise ValueError(message)
+
+    def request(self, method: str, path: str, **kwargs) -> requests.Response:
```

Comment:
I like having `request` as an option because it gives you full access to the tower API functionality, but just want to confirm that more convenient methods like `launch_workflow` from `sagetasks` will be a part of future work

Reply:
That's correct! We can add the convenience methods like `launch_workflow()` as we add the correponding function in `NextflowTowerOps`.